sqlmodel/main.py

```diff
@@ -475,6 +472,67 @@ def Relationship(
     return relationship_info
 
 
+def _merge_field_infos_and_update_class_dict(annotations, class_dict):
+    # Merge the FieldInfo into a single FieldInfo to prevent pydantic from dropping sqlmodel's attributes.
+    # Also update the class_dict if a class's default values include FieldInfo instances.
+    # Pydantic will automatically merge FieldInfo instances, but the merged object will be a `pydantic.FieldInfo`. All
+    # the sqlmodel-specific attributes are lost.
+    new_annotations = {}
+    updated = []
+    for attr, annotation in annotations.items():
+        # EARLY CONTINUE
+        if get_origin(annotation) is not Annotated:
+            new_annotations[attr] = annotation
+            continue
+        type_hint, *annotation_args = get_args(annotation)
+        # TODO: Check for other subclasses of `pydantic.FieldInfo`?
+        field_infos = [arg for arg in annotation_args if isinstance(arg, FieldInfo)]
+        # EARLY CONTINUE
+        if not field_infos:
+            # No FieldInfo in the annotation. Leave it as is.
+            new_annotations[attr] = annotation
+            continue
+        non_field_info_annotation_args = [
+            arg for arg in annotation_args if not isinstance(arg, FieldInfo)
+        ]
+        # If the default value is a FieldInfo, include that too.
+        default_field_info = None
+        if isinstance(class_dict.get(attr), FieldInfo):
+            default_field_info = class_dict.get(attr)
+            field_infos.append(default_field_info)
+
+        # HACK: Quick and dirty way of merging into a new sqlmodel.FieldInfo object.
+        with mock.patch("pydantic.fields.FieldInfo", FieldInfo):
```

Comment:
This call to mock.patch is no longer necessary. I'll implement the merging logic in sqlmodel before marking this PR as ready.

(I wanted to get feedback on the overall approach before taking the time to re-implement the pydantic logic.)